src/transformers/modeling_utils.py

```diff
@@ -3164,6 +3184,12 @@ def from_pretrained(
         if hasattr(model, "quantization_method"):
             model.is_quantized = True
 
+            # We store the original dtype for quantized models as we cannot easily retrieve them
+            # once the weights have been quantized
+            # Note that once you have loaded a quantized model, you can't change its dtype so this will
+            # remain a single source of truth
```

Comment:
Might be needed in the quantizer config?

Reply:
I think it is ok since users can always load back quantized models with new torch_dtype making that `_pre_quantization_dtype` obsolete